src/migration.ts

```diff
@@ -0,0 +1,90 @@
+import { PrismaClient } from "@prisma/client";
+import * as fs from 'fs';
+
+const prisma = new PrismaClient();
+//const fs = require('fs');
+
+async function readSpreadsheet(filePath: string) {
+    try {
+      const content = fs.readFileSync(filePath, 'utf-8');
+      const rows = content.split('\n').map(row => row.split(','));
+      //skip headers in row 1
+      const data = rows.slice(1)
+      await processExcelData(data)
+    } catch (error) {
+      console.log(error);
+    } finally {
+      await prisma.$disconnect();
+    }
+}
+
+async function processExcelData(data: string) {
+  try {
+    const createContactPromises = data.map(async (row) => {
+      const newTags = new Array();
+      for (let k = 0; k < 4; k++) {
```

Comment:
do we know that the tag list always has 4 items in it? if not, we will run into issues later on in the code. we may want to specifically extract the tags column and have special handling for it. also note that if needed, we can [get the current index in a map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#parameters)

Reply:
In the test file that UD sent us, the first four columns in their spreadsheet are titled "affiliation" and may contain a tag value, so each contact will have a maximum of 4 tags. Currently, the invalid tags (containing only whitespace) are filtered out later, and the valid tags are stored in a different array.